task/03-date-tasks.js

```diff
@@ -76,7 +85,31 @@ function isLeapYear(date) {
  *    Date(2000,1,1,10,0,0),  Date(2000,1,1,15,20,10,453)   => "05:20:10.453"
  */
 function timeSpanToString(startDate, endDate) {
-   throw new Error('Not implemented');
+   let time = endDate - startDate;
+   let milsec = parseInt(time % 1000);
+   let sec = Math.floor((time /1000) % 60);
+   let min = Math.floor((time /(1000*60))%60);
+   let hours = Math.floor((time / (1000*60*60)) %24);
+
+   if(startDate.getDay()!== endDate.getDay()){
+      hours = hours+24;
+   }
+   if(hours <10){
+       hours = "0" + hours;
+   }
+   if(min<10){
```

Comment:
it is necessary to observe indents.
Example: "if (min < 10) {"
https://javascript.info/coding-style